accepted/0000-replace-dwr.md

```diff
@@ -0,0 +1,74 @@
+  * <sup>*</sup> `SystemGroupsRenderer`: exposes the `renderAsync` method of the [`SystemGroupsRenderer`](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/action/renderers/SystemGroupsRenderer.java) class. It is [loaded](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/pages/yourrhn.jsp#L30) and [used](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/pages/yourrhn.jsp#L85) only in the `yourrhn.jsp` file.
+  * <sup>*</sup> `TasksRenderer`: exposes the `renderAsync` method of the [`TasksRenderer`](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/action/renderers/TasksRenderer.java) class. It is [loaded](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/pages/yourrhn.jsp#L15) and [used](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/pages/yourrhn.jsp#L48) only in the `yourrhn.jsp` file.
+
+  <sup>*</sup> All of these are only used in the [Your RHN](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/pages/yourrhn.jsp) page.
+
+## What is the expected outcome?
+SUMA working normally with the same AJAX requests being made and handled without any use of the DWR library.
+
+
+# Detailed design
+
+Although the solution could be a little risky, since it involves changes in a number of stable features, it is not so complex. The idea is to have a raw Servlet incorporated in SUMA working as backend for all the AJAX requests currently handled by the [DwrServlet](https://github.com/uyuni-project/uyuni/blob/master/java/code/webapp/WEB-INF/web.xml#L260). This servlet could be based on the URLs of the requests to determine which Java class to instantiate and which method to call on it.
+
+For the frontend side, the capabilities provided by DWR (both the utility functions from `util.js` file and the ajax calls) are replaceable using jQuery.
+
+There is a PoC in [this commit](https://github.com/wweellddeerr/uyuni/commit/9ddb68030dc49fa3272a062ceb7015c35b68d457) exploring how it is possible to use jQuery to make the ajax request and a raw Servlet as backend replacing the `DWRItemSelector` component of DWR. The SSM (one of the features that uses it) seems to work well after applying this patch.
+
+Using a raw Servlet as backend seems to be the best approach since (a) the methods currently exposed by DWR are only available through the `DwrServlet`, which is a raw servlet; and (b) these methods typically manipulate the HttpSession, making it difficult to use the HTTP API, for example.
```

Comment:
Can you elaborate on this comment a bit, pointing to what exactly we would need to manipulate and why?